gateway/src/apicast/backend_client.lua

```diff
@@ -168,6 +173,40 @@ function _M:authorize(...)
   return call_backend_transaction(self, auth_uri, authorize_options(using_oauth), ...)
 end
 
+local function add_transaction(transactions, index, cred_type, cred, reports)
+  transactions['transactions[' .. index .. '][' .. cred_type .. ']'] = cred
```

Comment:
We should use string.format for this to minimize string allocation.

Reply:
Right :+1: 
Fixed.